src/methodical/impl/combo/common.clj

```diff
@@ -8,7 +94,8 @@
   (when (seq primary-methods)
     (reduce
      (fn [next-method primary-method]
-       (with-meta (partial primary-method next-method) (meta primary-method)))
+       (u/fn-with-meta (partial* (u/unwrap-fn-with-meta primary-method) next-method)
```

Comment:
I wonder if maybe we should also have a `with-meta*` that does the right thing for functions but otherwise does normal `with-meta` and then have Kondo `:discouraged-var` tell everyone to use that. Then we don't need `fn-with-meta` and it's impossible to forget to use it

Reply:
Good idea!